protocol/adapters/datasync_client.go

```diff
@@ -0,0 +1,223 @@
+package adapters
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/sha256"
+	"encoding/binary"
+	"log"
+	"sort"
+	"time"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/gogo/protobuf/proto"
+	"github.com/status-im/mvds"
+	"github.com/status-im/status-console-client/protocol/v1"
+	whisper "github.com/status-im/whisper/whisperv6"
+)
+
+// DataSyncClient is an adapter for MVDS
+// that implements the Protocol interface.
+type DataSyncClient struct {
+	sync *mvds.Node
+	t    *DataSyncWhisperTransport
+}
+
+func NewDataSyncClient(sync *mvds.Node, t *DataSyncWhisperTransport) *DataSyncClient {
+	go sync.Run()
```

Comment:
it is a bit weird, that `NewDataSyncClient` mutates state of `sync` variable, that might be confusing. maybe require it to have a running node already?

you can always use type system to help you make stuff more explicit:
let's say you have `mvds.Node` struct, then you can have only one method there, `Run`, starting the goroutine and returning an object `*mvds.RunningNode` that actually have all the interface you need.

then this method will require `*mvds.RunningNode` and it will be pretty obvious.